### PR TITLE
Added new tests for deployments, added new resource archive.

### PR DESCRIPTION
**Pull request by hhovsepy**
Enabled and fixed disabled tests.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit a419d72d120dcc81d13db33c23080cf26c0a3199:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8a4715ed45f767c584bb14ab79bf2d86081d314d:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e712468a63b648bdd6075c9262f95419410638db:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/middleware/test_middleware_deployment.py [hunk 1]
```diff
@@ -298,8 +365,19 @@ def check_deployment_appears(provider, server, runtime_name):
             .format(runtime_name, server.name))
 
 
+def check_deployment_not_listed(provider, server, runtime_name):
+    provider.refresh_provider_relationships(method='ui')
+    wait_for(lambda: runtime_name not in
+             get_deployments_statuses(MiddlewareDeployment.deployments(provider=provider,
```
Comment (jaryn) [inline cfme/tests/middleware/test_middleware_deployment.py]: The indentation here should be improved.

**Comment by dajoRH** [comment]
Lint report for commit 095405ff1ecf8f3beb2e1050f0f67d85a6dceffa:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by hhovsepy** [comment]
@jaryn is there any other comment, or this PR can be merged?

**Pull request merged**